File: benchmarks/tsb/bench_range_index.ts
/**
 * Benchmark: RangeIndex construction, toArray(), slice(), contains()
 */
import { RangeIndex } from "../../src/index.js";

const N = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;

for (let i = 0; i < WARMUP; i++) {
  const r = new RangeIndex(N);
  r.toArray();
  r.slice(1000, 5000);
  r.contains(50_000);
}
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  const r = new RangeIndex(N);
  r.toArray();
  r.slice(1000, 5000);
  r.contains(50_000);
}
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "range_index",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_range_index.py
"""Benchmark: pd.RangeIndex construction, .tolist(), slice, contains on 100k"""
import json, time
import pandas as pd

N = 100_000
WARMUP = 3
ITERATIONS = 10

for _ in range(WARMUP):
    r = pd.RangeIndex(N)
    r.tolist()
    r[1000:5000]
    50_000 in r

start = time.perf_counter()
for _ in range(ITERATIONS):
    r = pd.RangeIndex(N)
    r.tolist()
    r[1000:5000]
    50_000 in r
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "range_index", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
